### PR TITLE
Add PN512 into target config

### PR DESCRIPTION
**Pull request by rajkan01**
- Add the PN512 NFC driver config in mbed_app.json to include PN512 driver source

_Note:_ 
@jamesbeyond manually tested this example in DISCO_L475VG target with PN512 NFC controller shield.

### Reviewers 
@0xc0170 @pan- @hugueskamba

### CODE REVIEW

**Review by pan- — CHANGES_REQUESTED** [review]
That is confusing, the `DISCO_L475VG_IOT01A` embed an NFC tag: the M24SR. It works out of the box with the EEPROM example but that NFC chip doesn't have what is required to run the SmartPoster example. 

I'd prefer to not have confusing default configurations in examples. With these lines it seems that the `DISCO_L475VG_IOT01A` is supported OOB when it's not.

**Comment by jamesbeyond** [comment]
Hi @pan-, I asked Raj to add that in to support `DISCO_L475VG_IOT01A`

The reason for that is for CI purpose, we need some common targets works with both the NFC example. currently `DISCO_L475VG_IOT01A` is good candidate. because we have not likely to have `NUCLEO_F401RE` in board farm.

secondly,  `DISCO_L475VG_IOT01A` supports `PN512` shields, with the shields it works just fine. So I don't think there will be a problem for `DISCO_L475VG_IOT01A` to support this example. Maybe we just need to be more clear on the Readme. says  `PN512` shields is a requirement to this example for both `DISCO_L475VG_IOT01A` and `NUCLEO_F401` even other targets as well

**Comment by pan-** [comment]
@jamesbeyond I do not understand what you try to solve here. We are very far from running any NFC example test in CI and `DISCO_L475VG_IOT01A` wouldn't be a reference target for testing the `PN512`. If it was a reference target, then we would have another problem which is the differentiation between `DISCO_L475VG_IOT01A ` boards with the `PN512` shield and those without (but that's an embarrassment of riches 😄 ). 

> So I don't think there will be a problem for DISCO_L475VG_IOT01A to support this example. Maybe we just need to be more clear on the Readme. says PN512 shields is a requirement to this example for both DISCO_L475VG_IOT01A and NUCLEO_F401 even other targets as well

The [README](https://github.com/ARMmbed/mbed-os-example-nfc/blob/master/NFC_SmartPoster/README.md) is already clear about it. I prefer users to fail the compilation then read the instructions to understand why it doesn't work rather than a successful compilation then error reported to the forum of this repo because the instructions where overlooked.

**Comment by jamesbeyond** [comment]
Hi @pan- 

>I do not understand what you try to solve here. 

The problem we want to resolve here is : we want to put NFC examples in CI and build with 'cmake' tools. but we need a pick a target can meet following requirements:
1. works with both examples.
2. currently support cmake.
3. possibly to be test for CI in future, (even it is embarrassment of riches 😄) 

For 1, obvious choices would be  "NUCLEO_F401RE" or "DISCO_L475VG_IOT01A".  but taking 2 and 3 into consideration,  `DISCO_L475VG_IOT01A` would the the preferred one, because `NUCLEO_F401RE` not support cmake tool at this stage, see [LINK](https://github.com/ARMmbed/mbed-tools/blob/master/docs/howto/mbed-tools-howto.md#mbed-os-configuration)

>The README is already clear about it. I prefer users to fail the compilation then read the instructions to understand why it doesn't work rather than a successful compilation then error reported to the forum of this repo because the instructions where overlooked (it compiles => it should work 🤷 ).

To respond to your proposal of let the complication failed, and force user to read the README, I'd say it is a double side blade. for some users it might works. but for others, they might be just go away simple blame the quality of the example and compilation failing, not bother about README at all. 

Thinking about the error message : 
```
CMakeFiles/NFC_SmartPoster.dir/main.cpp.obj -c ../main.cpp
../main.cpp:22:10: fatal error: nfc/controllers/PN512Driver.h: No such file or directory
```
Only if the user is well understood mbed-os, he then will know to add a `mbed_app.json` and set the flag of `"target.extra_labels_add": ["PN512"]`, these steps not mentioned anywhere in this readme. Other users might just go: WTH, where should I find these files
I'd rather let the build success, when user runs the binary, if detects targets not having the PN512 Shield installed, then print out a message in serial console told them to check the README.

Furthermore, if we make default build failing, it is also makes the CI harder to cope with it, if the CI not able to workaround it, then we have to take this example out of CI test, and future code change break this example would not be captured.

**Comment by 0xc0170** [comment]
If we shall not test `DISCO_L475VG_IOT01A`. We should test `NUCLEO_F401RE` for CMake. As I am checking CMake sources, we have `NUCLEO_F401RE` supported.

My proposal to unblock CMake: lets switch to `NUCLEO_F401RE`. I've tested it with CMake, it builds without issue (I recall we added the target to CMake long time ago).  If this is just for the build (if we dont have it in the farm, we have another problem). It might be the best to meet today. @rajkan01 is away, me and @hugueskamba can join instead. I'll schedule a meeting

File: NFC_SmartPoster/mbed_app.json [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {
     "target_overrides": {
+        "DISCO_L475VG_IOT01A": {
+            "target.extra_labels_add": ["PN512"]
```
Comment (hugueskamba) [inline NFC_SmartPoster/mbed_app.json:4]: Please explain this by amending the README.md

**Comment by rajkan01** [comment]
This PR change is redundant, as everyone agreed to use NUCLEO_F401RE board for this example.

**Pull request closed without merging**